### PR TITLE
Form Block: fixes for the CRM integration

### PR DESCRIPTION
**Pull request by jeherve**
Closes #16662 
Closes #16661 

#### Changes proposed in this Pull Request:

**Note**: this PR is open against #16658 and includes both #16662 and #16661. If we were to merge this in, we can close and ignore #16662 and #16661.

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* N/A

#### Testing instructions:

* Try to insert a form block and check the block sidebar in different scenarios:
    - Jetpack CRM not installed
    - Jetpack CRM installed but not active
    - Jetpack CRM active with version >4.0:
* Ensure the proper string is displayed every time.

#### Proposed changelog entry for your changes:

* N/A

### CODE REVIEW

File: extensions/blocks/contact-form/components/jetpack-crm-connection-settings.js
Comment (kbrown9) [inline extensions/blocks/contact-form/components/jetpack-crm-connection-settings.js:98]: It looks like I forgot to add the `jetpack-contact-form__crm_text` class to this element in my original commit. I'll fix that.

File: extensions/blocks/contact-form/components/jetpack-crm-connection-settings.js
Comment (kbrown9) [inline extensions/blocks/contact-form/components/jetpack-crm-connection-settings.js:74]: We need to return null when the CRM plugin is active and version is >4.0. We no longer display any text in that scenario, just the toggle. I'll push a commit that addresses this scenario.

Comment (kbrown9) [reply extensions/blocks/contact-form/components/jetpack-crm-connection-settings.js:74]: See commit 5e433a0.

File: extensions/blocks/contact-form/components/jetpack-crm-connection-settings.js
Comment (kbrown9) [reply extensions/blocks/contact-form/components/jetpack-crm-connection-settings.js]: See commit 5e433a0.

**Review by kbrown9 — APPROVED** [review]

**Comment by kbrown9** [comment]
Thanks for these changes! I've tested them, and everything looks good to me, except for a few things that I addressed in commit 5e433a0. Do you want to review that commit before we merge this into the `update/crm-text` branch?

**Comment by jeherve** [comment]
This looks good to me, thank you! Merging this now, and reviewing / approving the other one after that.

**Pull request merged**
